### ISSUE TITLE
Bug: "Cannot use destroyed queue" error when calling queue.destroy()

### ISSUE THREAD

**Issue by larsrickert** [Androz2091/discord-player#1106]
**Describe the bug**
When the queue contains some tracks and `queue.destroy()` is called a `Cannot use destroyed queue` error is emitted.

**Error reason**
Inside the `destroy` function the connection is ended/destroyed before `this.#destroyed` is set to true so the `this.connection.on("finish")` listener might be called before. Therefore the listeners `this.#watchDestroyed` check does not return true and the rest of the listener is executed. Eventually the `this.play()` method will be called but at this point `this.#destroyed` is true so an `Cannot use destroyed queue` error will be emitted.

**To Reproduce**
1. Add some tracksto the queue using `queue.addTracks()`
2. Register `player.on("error")` handler and log the error to see it
3. Wait for the discord bot to play a track
4. Call `queue.destory(true)`
5. Error from step 2 is being shown

**Expected behavior**
When calling `queue.destroy()` not `Cannot use destroyed queue` error should be emitted.

**Please complete the following information:**
 - Node Version: 16.13.2
 - Discord Player Version: ^5.2.2
 - Discord.js Version: ^13.6.0

**Comment by vanderrich** [comment]
had the same bug to my bot too

**Comment by larsrickert** [comment]
> had the same bug to my bot too

If you want a temp fix until PR #648 is merged, you can add the following to your `player.on("error")`:

```js
player.on('error', (queue, error) => {
  // temp fix until https://github.com/Androz2091/discord-player/pull/1107 is merged and released
  if (error.message === '[DestroyedQueue] Cannot use destroyed queue') return;

// rest of your code...
});
```
